### PR TITLE
dict_set container added

### PR DESCRIPTION
**Pull request by azhogin**
Also added PUSHSLICE_EMPTY for optimal creation of empty slice.

### CODE REVIEW

File: stdlib/cpp-sdk/tvm/dict_set.hpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    return succ;
+  }
+
+  void erase(key_type key) {
+    if (dict_.dictudel(key.get(), KeyLen))
+      --size_;
+  }
+
+  void insert(key_type key) {
+    // increase size if adding new key
```
Comment (akiramenai) [inline stdlib/cpp-sdk/tvm/dict_set.hpp:101]: It's quite obvious from the code.

File: stdlib/cpp-sdk/tvm/dict_set.hpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    bool right_end = v.is_end();
+    return left_end && right_end;
+  }
+  bool operator!=(dict_set_const_iterator v) const {
+    return !(*this == v);
+  }
+};
+
+template<unsigned KeyLen>
+class dict_set<schema::uint_t<KeyLen>> {
```
Comment (akiramenai) [inline stdlib/cpp-sdk/tvm/dict_set.hpp:59]: Shouldn't we disallow copying for now? Same for `dictionary`.

Comment (azhogin) [reply stdlib/cpp-sdk/tvm/dict_set.hpp:59]: Why? dict_set contains size and cell (dict), both are simple copyable values (with stack operations).

Comment (akiramenai) [reply stdlib/cpp-sdk/tvm/dict_set.hpp:59]: Ok then. So DUP is generated for a copy, isn't it?

Comment (azhogin) [reply stdlib/cpp-sdk/tvm/dict_set.hpp:59]: If you really perform copy, yes - DUP or PUSH for deeper element. If you just moving elements, no copies will be performed.

Comment (akiramenai) [reply stdlib/cpp-sdk/tvm/dict_set.hpp:59]: Ok, got it. Thanks.

**Pull request merged**